python/Ice/cancellation/server/chatbot.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) ZeroC, Inc.
+
+import Ice
+import asyncio
+# Slice module VisitorCenter in Greeter.ice maps to Python module VisitorCenter.
+import VisitorCenter
+
+class Chatbot(VisitorCenter.Greeter):
+    """
+    Chatbot is an Ice servant that implements Slice interface Greeter.
+    """
+
+    def __init__(self, delay: float):
+        """
+        Initializes a Chatbot servant with a delay.
+
+        Parameters
+        ----------
+        delay : float
+            The delay before returning each greeting.
+        """
+        self._delay = delay
+
+    # Implements the method greet from the Greeter class generated by the Slice compiler.
+    # This variant is the synchronous implementation.
+    async def greet(self, name: str, current: Ice.Current) -> str:
+        print(f"Dispatching greet request {{ name = '{name}' }}")
```

Comment:
Dispatch cancellation is not doable in Python?

Reply:
Doable yes. Not as elegant as with C# cancelation tokens. We can pass a future to the servant instead of a cancelation token.

Then instead of awaiting `asyncio.sleep` we would have to away both the sleep task and the extra future.